tfx/orchestration/experimental/kubernetes/kubernetes_dag_runner.py

```diff
@@ -0,0 +1,378 @@
+        container_image=_TFX_IMAGE,
+        command=_DRIVER_COMMAND + arguments,
+        container_name=container_name,
+        pod_labels={
+            'job-name': pod_label,
+        },
+    )
+    try:
+      batch_api.create_namespaced_job(
+          "default", job, pretty=True)
+    except client.rest.ApiException as e:
+      raise RuntimeError('Failed to submit job! \nReason: %s\nBody: %s' %
+                         (e.reason, e.body))
+
+    # Wait for pod to start
+    orchestrator_pods = []
+    core_api = kube_utils.make_core_v1_api()
+    start_time = datetime.datetime.utcnow()
+    while not orchestrator_pods and (datetime.datetime.utcnow() - start_time
+                                     ).seconds < 300:
+      try:
+        orchestrator_pods = core_api.list_namespaced_pod(
+            namespace='default',
+            label_selector='job-name={}'.format(pod_label)
+        ).items
+      except client.rest.ApiException as e:
+        if e.status != 404:
+          raise RuntimeError('Unknown error! \nReason: %s\nBody: %s' %
+                             (e.reason, e.body))
+      time.sleep(1)
+
+    # transient orchestrator should only have 1 pod
+    orchestrator_pod = orchestrator_pods.pop()
+    pod_name = orchestrator_pod.metadata.name
+
+    #TODO(https://github.com/tensorflow/tfx/pull/2248): refactor protected members
+    cond = kubernetes_component_launcher._pod_is_not_pending # pylint: disable=W0212
+    absl.logging.info('Waiting for pod "default:%s" to start.', pod_name)
+    kube_utils.wait_pod(
+        core_api,
+        pod_name,
+        'default',
+        exit_condition_lambda=cond,
+        condition_description='non-pending status')
+
+    # stream logs from pod
+    absl.logging.info('Start log streaming for pod "default:%s".', pod_name)
+    try:
+      logs = core_api.read_namespaced_pod_log(
+          name=pod_name,
+          namespace='default',
+          container=container_name,
+          follow=True,
+          _preload_content=False).stream()
+    except client.rest.ApiException as e:
+      raise RuntimeError(
+          'Failed to stream the logs from the pod!\nReason: %s\nBody: %s' %
+          (e.reason, e.body))
+
+    for log in logs:
+      absl.logging.info(log.decode().rstrip('\n'))
+
+    cond = kubernetes_component_launcher._pod_is_done # pylint: disable=W0212
+    resp = kube_utils.wait_pod(
+        core_api,
+        pod_name,
+        'default',
+        exit_condition_lambda=cond,
+        condition_description='done state',
+        expotential_backoff=True)
+
+    if resp.status.phase == kube_utils.PodPhase.FAILED.value:
+      raise RuntimeError('Pod "default:%s" failed with status "%s".' %
+                         (pod_name, resp.status))
+
+  def _serialize_pipeline(self, pipeline: tfx_pipeline.Pipeline) -> Text:
+    """Serializes a TFX pipeline.
+
+    To be replaced with the "portable core" of the unified TFX orchestrator:
+    https://go/tfx-dsl-ir
```

Comment:
Please describe the serialization format specifically enough so that someone else could implement this method.